lib/request.js

```diff
@@ -63,22 +63,37 @@ class Request extends AsyncResource {
     if (!(typeof opts.path === 'string' && opts.path[0] === '/')) {
       throw new Error('path must be a valid path')
     }
-    this.method = opts.method
 
     if (!(typeof opts.method === 'string' && methods[opts.method])) {
       throw new Error('method must be a valid method')
     }
-    this.path = opts.path
 
     if (!isValidBody(opts.body)) {
       throw new Error('body must be a string, a Buffer or a Readable stream')
     }
+
+    this.method = opts.method
+
+    this.path = opts.path
+
     this.body = opts.body
 
-    this.headers = opts.headers
+    this.host = opts.headers && (opts.headers.host || opts.headers.Host)
+
+    this.chunked = !opts.headers || opts.headers['content-length'] === undefined
```

Comment:
linter complained about `hasOwnProperty`, I think `=== undefined` should be good enough.